### PR TITLE
42 profile directive part1

### PR DESCRIPTION
**Pull request by dshcheung**
please merge #50 before checking this one as i need to rebase

### CODE REVIEW

File: app/assets/javascripts/angular/controllers/summoner_controller.js.erb [hunk 1]
```diff
@@ -4,12 +4,12 @@ app.controller('SummonerCtrl', ['$scope', '$routeParams', '$http',function($scop
   // Summoner Information
   // Set Initial Information
   $scope.summonerInfo = {};
+  $scope.summonerInfo.searchData = {
+    'id': $routeParams.summonerId,
+    'region': $scope.$parent.selectedRegion.code
```
Comment (h8rry) [inline app/assets/javascripts/angular/controllers/summoner_controller.js.erb]: Why do we need to use `$parent` in `$parent.selectedRegion`?

File: app/assets/templates/partials/profileTabPartial.html [hunk 2]
```diff
@@ -1 +1,103 @@
-<h1>testing profile</h1>
```
Comment (h8rry) [inline app/assets/templates/partials/profileTabPartial.html:1]: shouldn't use camelCase for the file name

File: app/assets/stylesheets/profileTabPartial.css.scss [hunk 3]
```diff
@@ -1,4 +1,67 @@
 @import "variables";
```
Comment (h8rry) [inline app/assets/stylesheets/profileTabPartial.css.scss:1]: shouldn't use camelCase for the file name

File: app/controllers/summoners_controller.rb [hunk 4]
```diff
@@ -50,4 +49,113 @@ def find_by_name
...
+      if response.empty?
+        render json: {success: true, league_entry: nil}, status: 200
+      else
+        for i in 0..response["#{params['id']}"].length - 1
+          league_name = response["#{params['id']}"][i]['queue']
+          tier = response["#{params['id']}"][i]['tier'].capitalize
+          division = response["#{params['id']}"][i]['entries'][0]['division']
+          if league_name[/(SOLO)/]
+            league[:solo5] = response["#{params['id']}"][i]
+            league[:solo5]["badge_icon"] = ActionController::Base.helpers.asset_path("badge" +  tier + division + ".png")
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb]: you can consistently use `#{}` for string concatenation

File: app/controllers/summoners_controller.rb [hunk 5]
```diff
@@ -50,4 +49,113 @@ def find_by_name
...
+      url = "https://#{params['region']}.api.pvp.net/api/lol/#{params['region']}/v1.3/stats/by-summoner/#{params['id']}/summary?season=SEASON2015&api_key=#{ENV['API_KEY']}"
+      response = JSON.parse(open(url).read)
+      if response.empty?
+        render json: {success: true, stats: stats}, status: 200
+      else
+        game = response["playerStatSummaries"]
+        for i in 0..game.length - 1
+          game_type = response["playerStatSummaries"][i]["playerStatSummaryType"]
+          puts "#{i}"
+          if game_type == "RankedTeam5x5"
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb:138]: You can use `case` function instead

File: app/controllers/summoners_controller.rb [hunk 6]
```diff
@@ -50,4 +49,113 @@ def find_by_name
...
+      team3: nil
+    }
+
+    @tries = 0
+    begin
+      url = "https://#{params['region']}.api.pvp.net/api/lol/#{params['region'].downcase}/v2.5/league/by-summoner/#{params['id']}/entry?api_key=#{ENV['API_KEY']}"
+      response = JSON.parse(open(url).read)
+      if response.empty?
+        render json: {success: true, league_entry: nil}, status: 200
+      else
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb]: Seems like the code is very "imperative", meaning not very modular. You can try modularizing it if you have time

File: app/assets/templates/partials/profileTabPartial.html (hunk 2 above)
Comment (dshcheung) [inline app/assets/templates/partials/profileTabPartial.html:1]: will change on another branch

File: app/assets/stylesheets/profileTabPartial.css.scss (hunk 3 above)
Comment (dshcheung) [inline app/assets/stylesheets/profileTabPartial.css.scss:1]: will change on another branch

File: app/controllers/summoners_controller.rb (hunk 5 above)
Comment (dshcheung) [inline app/controllers/summoners_controller.rb:138]: will change on another branch

**Pull request merged**